src/datasources/balances-api/entities/__tests__/asset-price.schema.spec.ts

```diff
@@ -0,0 +1,19 @@
+import { AssetPriceSchema } from '@/datasources/balances-api/entities/asset-price.entity';
+import { faker } from '@faker-js/faker';
+
+describe('AssetPriceSchema', () => {
+  it('should allow an object with string keys and a mixture of number and null values', () => {
+    const assetPrice = {
+      [faker.finance.ethereumAddress()]: {
+        [faker.finance.currencyCode()]: faker.helpers.arrayElement([
+          faker.number.float(),
+          null,
+        ]),
+      },
+    };
+
+    const result = AssetPriceSchema.safeParse(assetPrice);
+
+    expect(result.success).toBe(true);
+  });
```

Comment:
I think we should include at least one test in which the validation not passing, wdyt?

Reply:
Added in 369edc9f04839313571bd5f137e44da0e49d1abd.